src/OpenFeature.Contrib.Providers.Flagd/Resolver/InProcess/CustomEvaluators/FractionalEvaluator.cs

```diff
@@ -0,0 +1,100 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using JsonLogic.Net;
+using Murmur;
+using Newtonsoft.Json.Linq;
+using Semver;
+
+namespace OpenFeature.Contrib.Providers.Flagd.Resolver.InProcess.CustomEvaluators
+{
+    /// <inheritdoc/>
+    public class FractionalEvaluator
+    {
+
+        class FractionalEvaluationDistribution
+        {
+            public string variant;
+            public int percentage;
+        }
+
+        internal object Evaluate(IProcessJsonLogic p, JToken[] args, object data)
+        {
+            // check if we have at least two arguments:
+            // 1. the property value
+            // 2. the array containing the buckets
+
+            if (args.Length == 0)
+            {
+                return null;
+            }
+
+            var flagdProperties = new FlagdProperties(data);
+
+            // check if the first argument is a string (i.e. the property to base the distribution on
+            var propertyValue = flagdProperties.TargetingKey;
+            var bucketStartIndex = 0;
+
+            var arg0 = p.Apply(args[0], data);
+
+            if (arg0 is string stringValue)
+            {
+                propertyValue = stringValue;
+                bucketStartIndex = 1;
+            }
+
+            var distributions = new List<FractionalEvaluationDistribution>();
+
+            for (var i = bucketStartIndex; i < args.Length; i++)
+            {
+                var bucket = p.Apply(args[i], data);
+
+                if (!bucket.IsEnumerable())
+                {
+                    continue;
+                }
+
+                var bucketArr = bucket.MakeEnumerable().ToArray();
+
+                if (bucketArr.Count() < 2)
+                {
+                    continue;
+                }
+
+                if (!bucketArr.ElementAt(1).IsNumeric())
+                {
+                    continue;
+                }
+
+
+                distributions.Add(new FractionalEvaluationDistribution
+                {
+                    variant = bucketArr.ElementAt(0).ToString(),
+                    percentage = Convert.ToInt32(bucketArr.ElementAt(1))
```

Comment:
Minor - along with the calculation, please consider checking if percentage add upto 100%. This is needed as per our spec [1] and see Java implementation for reference [2]

[1] - https://github.com/open-feature/flagd/blob/main/docs/reference/specifications/custom-operations/fractional-operation-spec.md

[2] - https://github.com/open-feature/java-sdk-contrib/blob/main/providers/flagd/src/main/java/dev/openfeature/contrib/providers/flagd/resolver/process/targeting/Fractional.java#L67-L69